cmd/ipi-deprovision/ipi-deprovision.sh

```diff
@@ -29,7 +29,24 @@ handle_cluster () {
   ./bin/hiveutil aws-tag-deprovision "${cluster_name}=owned" --region "${region}"
 }
 
-
+collect_metadata () {
+  local cluster_name
+  cluster_name=$1
+  ###eg cluster_name=kubernetes.io/cluster/ci-op-724qy8fn-55c01-s5c8w
+  local ns
+  ns=${cluster_name:22:14}
+  echo "cluster is used by namespace ${ns}"
+  echo "oc get pod -n ${ns}"
+  local cache_json
+  cache_json=$(oc get pods -n "${ns}" -o json)
+  ###eg, the value of env var CLUSTER_NAME: ci-op-724qy8fn-55c01
+  for pod_name in $(echo "${cache_json}" | jq --arg prefix "${ns}-" -r '.items[] | . as $pod | $pod.spec.containers[] | select(.name == "setup") | .env[]? | select(.name == "CLUSTER_NAME" and (.value | startswith($prefix) )) | $pod.metadata.name')
```

Comment:
Why check the prefix when we know exactly the value of the cluster name to check?

Reply:
Totally true.